clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestFuture.java

```diff
@@ -83,15 +87,17 @@ public boolean failed() {
      * @return true if it is retriable, false otherwise
      */
     public boolean isRetriable() {
-        return exception instanceof RetriableException;
+        return exception() instanceof RetriableException;
```

Comment:
I think it's actually better this way since `isRetriable` doesn't make sense unless the request has failed.